rustls/src/conn.rs

```diff
@@ -1356,7 +1331,22 @@ impl Quic {
     }
 }
 
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub(crate) enum Side {
+    Client,
+    Server,
+}
+
+pub(crate) use sealed::SealedSideData;
+
 /// Data specific to the peer's side (client or server).
-pub trait SideData {}
+pub trait SideData: SealedSideData {}
+
+mod sealed {
```

Comment:
The point of this was to keep the associated const crate private. We can't put it in `Sealed` because that's used for other traits as well. We could put it in a separate trait that's directly `pub(crate)` and then use both a private supertraits for `SideData`, but that doesn't seem like a significant improvement.